repo/fsrepo/migrations/ipfsfetcher/ipfsfetcher_test.go

```diff
@@ -0,0 +1,97 @@
+package ipfsfetcher
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"testing"
+)
+
+var runIpfsTest = flag.Bool("ipfstest", false, "Run IpfsFetcher tests")
+
+func TestIpfsFetcher(t *testing.T) {
+	if !*runIpfsTest {
+		t.Skip("manually-run dev test, use '-ipfstest' flage to run")
```

Comment:
Is the reason this test is manual because it makes network requests?

Reply:
Yes, and I did not know a good way to simulate or mock an IPFS node, but I wanted to have some test that a developer could run to check if things are working.  WDYT? Leave it, toss it, create a better test?

Reply:
There are some tests in the integration test package (e.g. https://github.com/ipfs/go-ipfs/blob/master/test/integration/three_legged_cat_test.go) that could be of inspiration here.

If it's a pain to write the test lmk and we can decide on an alternative. Using a flag/env var seems fine (I'm not sure if we have an appropriate one, although perhaps this one https://github.com/ipfs/go-ipfs/blob/ef866a1400b3b2861e5e8b6cc9edc8633b890a0a/test/integration/addcat_test.go#L173 would do), although I don't really trust tests that don't run in CI since they aren't reliably run before merging PRs and are subject to breaking.

@Stebalien lmk if you have any additional thoughts here given you've been thinking about our CI situation a bit, or it's just 👍.

Reply:
We need to test this in CI but we can use sharness tests. If a test is disabled in CI, it might as well not exist.

Reply:
@aschmahmann I will see what I can do with sharness, but likely this is going to have to become a job for your team if it needs to get done before the upcoming release.